spec/fixtures/custom_nginx.template

```diff
@@ -312,60 +312,157 @@ http {
-                      ngx.log(ngx.INFO, "client ponged")
-
-                    elseif typ == "text" then
-                      local bytes, err = wb:send_text(data)
-                      if not bytes then
-                        ngx.log(ngx.ERR, "failed to send text: ", err)
-                        return ngx.exit(444)
-                      end
-                    end
+        location = /post {
+            access_by_lua_block {
+                local mu = require("spec.fixtures.mock_upstream")
+                return mu.filter_access_by_method("POST")
+            }
+            content_by_lua_block {
+                local mu                 = require("spec.fixtures.mock_upstream")
+                local headers            = ngx.req.get_headers()
+                local data, form, params = mu.parse_post_data(headers)
+                return mu.send_json_response({
+                    args    = ngx.req.get_uri_args(),
+                    data    = data,
+                    form    = form,
+                    headers = headers,
+                    origin  = ngx.var.remote_addr,
+                    params  = params,
+                    url     = mu.get_request_url(),
+                })
+            }
+        }
 
-                  else
-                    local bytes, err = wb:send_ping()
-                    if not bytes then
-                      ngx.log(ngx.ERR, "failed to send ping: ", err)
-                      return ngx.exit(444)
-                    end
-                  end
+        location ~ "^/basic-auth/(?<username>[a-zA-Z0-9_]+)/(?<password>[.]+)$" {
+            access_by_lua_block {
+                local mu = require("spec.fixtures.mock_upstream")
+                return mu.filter_access_by_basic_auth(ngx.var.username,
+                                                      ngx.var.password)
+            }
+            content_by_lua_block {
+                local mu = require("spec.fixtures.mock_upstream")
+                return mu.send_json_response({
+                  authenticated = true,
+                  user          = ngx.var.username,
+                })
+            }
+        }
+
+        location ~ "^/(request|anything)$" {
+            content_by_lua_block {
+                local mu                 = require("spec.fixtures.mock_upstream")
+                local headers            = ngx.req.get_headers()
+                local data, form, params = mu.parse_post_data(headers)
+
+                return mu.send_json_response({
+                    args    = ngx.req.get_uri_args(),
+                    data    = data,
+                    form    = form,
+                    headers = headers,
+                    method  = ngx.var.request_method,
+                    origin  = ngx.var.remote_addr,
+                    params  = params,
+                    url     = mu.get_request_url(),
+                })
+            }
+        }
+
+        location ~ "^/delay/(?<delay_seconds>\d{3})$" {
```

Comment:
This regexp will only match delays of 3 digits, which means requests would have to specify `/delay/001` 😅 

Will replace with `{1,3}`.